internal/sources/api_upstream_test.go

```diff
@@ -245,39 +248,228 @@ info:
 	Describe("FetchRegistry", func() {
 		var registryConfig *config.RegistryConfig
 
-		BeforeEach(func() {
-			registryConfig = &config.RegistryConfig{
-				Name: "test-registry",
-				API:  &config.APIConfig{},
-			}
+		Context("Successful fetch with single page", func() {
+			BeforeEach(func() {
+				mockServer = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+					if r.URL.Path == serversAPIPath {
+						w.Header().Set("Content-Type", "application/json")
+						w.WriteHeader(http.StatusOK)
+						_, _ = w.Write([]byte(`{
+							"servers": [
+								{
+									"server": {
+										"name": "test-server",
+										"description": "A test server"
+									},
+									"_meta": {}
+								}
+							],
+							"metadata": {
+								"nextCursor": "",
+								"count": 1
+							}
+						}`))
+					} else {
+						w.WriteHeader(http.StatusNotFound)
+					}
+				}))
+				registryConfig = &config.RegistryConfig{
+					Name:   "test-registry",
+					Format: config.SourceFormatUpstream,
+					API: &config.APIConfig{
+						Endpoint: mockServer.URL,
+					},
+				}
+			})
+
+			It("should fetch servers successfully", func() {
+				result, err := handler.FetchRegistry(ctx, registryConfig)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).NotTo(BeNil())
+				Expect(result.ServerCount).To(Equal(1))
+				Expect(result.Format).To(Equal(config.SourceFormatUpstream))
+				Expect(result.Hash).NotTo(BeEmpty())
+				Expect(result.Registry).NotTo(BeNil())
+				Expect(result.Registry.Data.Servers).To(HaveLen(1))
+				Expect(result.Registry.Data.Servers[0].Name).To(Equal("test-server"))
+			})
+		})
+
+		Context("Successful fetch with pagination", func() {
```

Comment:
Is this testing pagination? We're testing that the right things comes back if we specify page2, but not testing how many servers are on a page and that the expect number of pages exist?